result-watcher.bal

```diff
@@ -214,6 +278,7 @@ public isolated class ResultWatcher {
                     }
                     if newInterval == () {
                         check self.unschedule();
+                        _ = check removeResultWatcherFromRegistry(self.stepId); // not sure if this is needed here
```

Comment:
The final unschedule should remove the watcher completely from memory (that also means from the registry).